roles/step_ca/tasks/check.yml

```diff
@@ -19,6 +20,25 @@
     step_ca_existing_root_file is defined or
     step_ca_existing_key_file is defined
 
+- name: Verify existing root key/password
+  assert:
+    that:
+      - step_ca_existing_key_file | length > 0
+    fail_msg: Existing root key must be passed if its password is defined
+  when: step_ca_existing_key_password is defined
+
+# TODO: Remove deprecation warning and add assert in `Verify existing root cert/key`
+- name: Check that password is provided
+# Deprecation warning 'hack'
+  ignore_errors: yes
+  fail:
+    msg: >-
+      [Warning] Root key password is not provided for remote key. This will later be disallowed in future versions.
+      Either use unencrypted local (on controller) key, or use an encrypted key, either local or remote.
+  when: >
+    step_ca_existing_key == 'remote' and
+    step_ca_existing_key_password is not defined
```

Comment:
Instead of warning people with a "hacky" fail task, I think a WARNING block in the documentation would work better here. Most people will only want to initialize their CA once and will read the accompanying docs to achieve that. If they still decide that using an unencrypted root key is a good choice, then that's not our call to make. I'll move that warning to the README

Reply:
I was considering that for people who are running it automatically, like if they want to update `step-ca`. They would not be look at the documentation just for that, but if they get a red error (non-blocking) message, that would raise their attention

Reply:
But if they already have a ca initialized, the entire code path that initializes the CA (including copying all the keys and so on) is never called and so we never even touch the keys. They could have long removed the root key from the CA and this check would still fire just because they still have the old role variables set.

Don't get me wrong, I understand your intent here. People really shouldn't store root ca keys in plaintext. I just don't think we are in a very good position to enforce that, nor do I want to.

Reply:
Oh yeah, I forgot to consider that. I am ok with it like this then